app/models/feed.rb

```diff
@@ -58,7 +90,11 @@ def default_runtime_settings?
   end
 
   def published_url
-    "#{podcast.base_published_url}/#{published_path}"
+    if private?
+      "#{podcast.base_private_url}/#{published_path}{?auth}"
+    else
+      "#{podcast.base_published_url}/#{published_path}"
+    end
```

Comment:
As far as I could tell, this `feed.published_url` wasn't used anywhere yet.  So should be an issue to vary it based on public vs private feeds.